src/PaymentServer/Processors/Stripe.hs

```diff
@@ -83,3 +126,56 @@ webhook d Event{eventId=Just (EventId eventId), eventType=ChargeSucceededEvent,
 webhook d _ =
   -- TODO: Record the eventId somewhere.
   return Ok
+
+
+-- | Browser facing API that takes token, voucher and a few other information
+-- and calls stripe charges API. If payment succeeds, then the voucher is stored
+-- in the voucher database.
+type ChargesAPI = "charge" :> ReqBody '[JSON] Charges :> Post '[JSON] Acknowledgement
+
+data Charges = Charges
+  { token :: Text
+  , voucher :: Voucher
+  , amount :: Int
```

Comment:
This is in the smallest unit of the selected currency, right?  So for USD it gives a number of cents?

Reply:
Yes, that is what it looks like.